File: examples/java-matter-controller/java/src/com/matter/controller/commands/common/IPAddress.java
package com.matter.controller.commands.common;

import java.net.InetAddress;

public final class IPAddress {
  private InetAddress address;

  public IPAddress(InetAddress address) {
    this.address = address;
  }

  public void setAddress(InetAddress address) {
    this.address = address;
  }

  @Override
  public String toString() {
    return address.toString();
  }
}

File: examples/java-matter-controller/java/src/com/matter/controller/commands/common/IPAddress.kt
/*
 * SPDX-FileCopyrightText: 2023 Project CHIP Authors
 *
 * SPDX-License-Identifier: Apache-2.0
 *
 */
package com.matter.controller.commands.common

import java.net.InetAddress

class IPAddress(private var address: InetAddress) {
  fun setAddress(address: InetAddress) {
    this.address = address
  }

  fun getHostAddress(): String {
    return address.hostAddress
  }

  override fun toString(): String {
    return address.toString()
  }
}
